sdk/test-utils/perfstress/src/program.ts

```diff
@@ -0,0 +1,334 @@
+ * then gets the test ready for a performance/stress test run.
+ *
+ * Use it like:
+ *
+ * ```ts
+ * export class Delay500ms extends PerfStressTest<string> {
+ *   public options = {};
+ *   async run(): Promise<void> {
+ *     await delay(500);
+ *   }
+ * }
+ *
+ * const perfStressProgram = new PerfStressProgram(Delay500ms);
+ *
+ * perfStressProgram.run();
+ * ```
+ */
+export class PerfStressProgram {
+  private testName: string;
+  private options: PerfStressOptionDictionary<DefaultPerfStressOptionNames>;
+  private parallelNumber: number;
+  private tests: PerfStressTest<string>[];
+
+  /**
+   * Receives a test class to instantiate and execute.
+   * Parses the test's options and creates some shortcuts.
+   * It will instantiate the same test class as many times as the "parallel" command line parameter specifies,
+   * which defaults to 1.
+   *
+   * @param testClass The testClass to be instantiated.
+   */
+  constructor(testClass: PerfStressTestConstructor<string>) {
+    this.testName = testClass.name;
+    this.options = parsePerfStressOption(defaultPerfStressOptions);
+    this.parallelNumber = Number(this.options.parallel.value);
+
+    console.log(`=== Creating ${this.parallelNumber} instance(s) of ${this.testName} ===`);
+    this.tests = new Array<PerfStressTest<string>>(this.parallelNumber);
+
+    for (let i = 0; i < this.parallelNumber; i++) {
+      const test = new testClass();
+      test.parseOptions();
+      this.tests[i] = test;
+    }
+  }
+
+  /**
+   * Does some calculations based on the parallel executions provided,
+   * then logs them in a friendly way.
+   *
+   * In languages supporting threads, "parallels" mean new threads created in which we run
+   * as many functions as possible. Keep in mind that we currently don't support multiple threads in NodeJS.
+   * We might be using workers eventually, but for now, "parallel" executions are
+   * promises that are executed one after the other without waiting for the previous one to finish.
+   * We wait for all of the promises to resolve to consider the "parallel" execution finished.
+   *
+   * The information logged consists of:
+   *
+   * - The total operations executed on that parallel run.
+   * - The operations per second, which is made through adding up each one of the parallel's
+   *   completed operations, divided by the seconds elapsed.
+   * - Seconds per operation, which is 1 / operationsPerSecond.
+   * - An average of the total operations by the operations per second, which we call weighted-average.
+   *
+   * @param parallels Parallel executions
+   */
+  private logResults(parallels: PerfStressParallel[]): void {
+    const totalOperations = parallels.reduce((sum, i) => sum + i.completedOperations, 0);
+    const operationsPerSecond = parallels.reduce((sum, parallel) => {
+      return sum + parallel.completedOperations / (parallel.lastMillisecondsElapsed / 1000);
```

Comment:
```suggestion
      return sum + (parallel.completedOperations / (parallel.lastMillisecondsElapsed / 1000));
```

Reply:
This will automatically be replaced after running `rushx format`. The idea of automatic formatting is to not have to think on formatting.